pkg/kubelet/kubelet.go

```diff
@@ -932,12 +929,20 @@ func (kl *Kubelet) pullImage(img string, ref *api.ObjectReference) error {
 
 // Kill all running containers in a pod (includes the pod infra container).
 func (kl *Kubelet) killPod(runningPod kubecontainer.Pod) error {
+
 	// Send the kills in parallel since they may take a long time.
 	errs := make(chan error, len(runningPod.Containers))
 	wg := sync.WaitGroup{}
 	for _, container := range runningPod.Containers {
 		wg.Add(1)
 		go func(container *kubecontainer.Container) {
+			// Call the networking plugin for teardown.
```

Comment:
@pmorie I think a bigger question to this is whether we want the kubelet knows those 'synthetic' containers. Should we hide it under the container runtime? Should it even be included in the `runningPod`?
If we just let the container runtime handle the infra container, it might do some killing/restarting out of the kubelet's control.